src/api/services/article/article.service.ts

```diff
@@ -0,0 +1,74 @@
+import type { CustomError } from "../../types/error.js";
+import * as ArticleRepository from "../../repositories/article.repository.js";
+import type { ArticleDto } from "./article.dto.js";
+import type { FindManyParams } from "../../types/express.d.ts";
+
+const ArticleService = {
+  async createArticle(articleData: ArticleDto, userId: number) {
+    const newArticle = await ArticleRepository.create({
+      ...articleData,
+      user: { connect: { id: userId } },
+    });
+    return newArticle;
+  },
+
+  async findUniqueArticle(articleId: number, userId?: number) {
+    const article = await ArticleRepository.findById(articleId);
+
+    if (!article) {
+      throw new Error("존재하지 않는 게시글입니다.");
+    }
+
+    if (!userId) {
+      return { ...article, isLiked: false };
+    }
+
+    // 좋아요 정보 조회
+    const like = await ArticleRepository.findLikeByUserAndArticle(userId, articleId);
+
+    return { ...article, isLiked: !!like };
+  },
+
+  async updateArticle(id: number, updateData: ArticleDto, userId: number) {
+    const article = await ArticleRepository.findById(id);
+
+    if (!article) {
+      const error: CustomError = new Error("존재하지 않는 게시글입니다.");
+      error.statusCode = 404;
+      throw error;
+    }
+
+    if (article.userId != userId) {
```

Comment:
느슨한 비교 말고 strict equal로 체크해주세요.
```suggestion
    if (article.userId !== userId) {
```